rmgpy/rmg/input.py

```diff
@@ -1613,7 +1666,7 @@ def save_input_file(path, rmg):
     if rmg.species_constraints:
         f.write('generatedSpeciesConstraints(\n')
         for constraint, value in sorted(list(rmg.species_constraints.items()), key=lambda constraint: constraint[0]):
-            if value is not None:
+            if value is not None and constraint is not "explicitlyAllowedMolecules":
```

Comment:
Your commit message on this change suggests you intended to review it. Just checking it doesn't get forgotten.